### DISCUSSION TITLE
Support for apollo federation 2.5?

### DISCUSSION THREAD

**Discussion by nicolasspilman** [Netflix/dgs-framework#1806] [General]
Thanks for all the work on this project.  It's fantastic.

Are their plans to support apollo federation 2.5 on the roadmap?  Looks like the current version locks the federation-graphql-java-support library at 3.0.0, while the latest version 4.3.0 includes the 2.5 spec.  

I tested a fork locally where I updated the version to 4.3.0 and it worked for my initial tests.

https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs-platform/build.gradle.kts#L76
```
     api("com.apollographql.federation:federation-graphql-java-support") {
            version {
                strictly("[3.0.0, 4.3.0]")
                prefer("4.3.0")
            }
        }
```
Are there any bits in this code base that would be affected by a version update like this, or does this library defer all federation related code to the library above?

Thanks for any info you can provide.

**Comment by paulbakker** [comment]
As long as the API of the library didn't change it should just work (which it seems to do). 

@srinivasankavitha any reason it's not on the new version yet?

**Comment by srinivasankavitha** [comment]
No reason to not update, we will be switching in the next release. Sent from my iPhoneOn Feb 1, 2024, at 3:44 PM, Paul Bakker ***@***.***> wrote:﻿
As long as the API of the library didn't change it should just work (which it seems to do).
@srinivasankavitha any reason it's not on the new version yet?

**Comment by nicolasspilman** [comment]
That's awesome.  Thanks for the quick responses.
